source/libraries/ContractDeployer.ts

```diff
@@ -148,7 +148,7 @@ Deploying to: ${networkConfiguration.networkName}
         if (contract.relativeFilePath.startsWith('legacy_reputation/')) return;
         if (contractName !== 'Map' && contract.relativeFilePath.startsWith('libraries/')) return;
         // Check to see if we have already uploded this version of the contract
-        if (await this.shouldUpgradeContract(contract, contractsToDelegate[contractName])) {
+        if (typeof this.configuration.controllerAddress !== "undefined" && await this.shouldUpgradeContract(contract, contractsToDelegate[contractName])) {
```

Comment:
Not related to this change but it actually looks like this is doing the opposite of what it says if I'm reading correctly. looks like if `shouldUpgradeContract` returns true it tries to reuse the existing contract. otherwise looks fine though

Reply:
That's the intended behavior.

"Upgrade" means "only upload contracts that have changed", or else we just upload everything.

Reply:
Totally. That is a terribly named function!